extensions/sql-database-projects/src/models/project.ts

```diff
@@ -553,7 +573,11 @@ export class Project implements ISqlProject {
 
 	public createFileProjectEntry(relativePath: string, entryType: EntryType, sqlObjectType?: string): FileProjectEntry {
 		let platformSafeRelativePath = utils.getPlatformSafeFileEntryPath(relativePath);
-		return new FileProjectEntry(Uri.file(path.join(this.projectFolderPath, platformSafeRelativePath)), relativePath, entryType, sqlObjectType);
+		return new FileProjectEntry(
+			Uri.file(path.join(this.projectFolderPath, platformSafeRelativePath)),
+			utils.convertSlashesForSqlProj(relativePath),
```

Comment:
Technically I could fix our problem without adding `utils.convertSlashesForSqlProj` call here, as I already updated `addScriptItem` and `addFolderItem` to normalize the path for comparison, but I felt like this would make it a little bit less error-prone in case there are other places in the future calling this.

I also considered adding this normalization in the `FileProjectEntry` constructor directly, to solve the problem once and for all, but after a closer look I assumed that `FileProjectEntry` was designed as DTO, so I didn't want to break that pattern by introducing logic in there.

Reply:
There is actually another issue here as well which I did not address - trailing `\`. i.e. `folder1` vs. `folder1\`. It will not surface if you only use `addToProject`, but if you use `addFolderItem` or create an entry using this factory method directly, then you may hit it. We can update `convertSlashesForSqlProj` helper to trim `\` maybe? I think the purpose of the method was to normalize the values for `sqlproj` file, right?